### PR TITLE
Improvements of pre-handle workflow

### PR DESCRIPTION
**Pull request by netopyr**
This PR implements suggestions that were made in the PR about updates in the pre-handle workflow.

### CODE REVIEW

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

File: hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/test/meta/TransactionMetadataTest.java
Comment (devnoh) [inline hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/test/meta/TransactionMetadataTest.java:55]: I think it needs some clean formatting here.

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Pull request merged**
